### PR TITLE
feat(VISE-CPC-496): Get a Single Visit by Visit ID

### PR DESCRIPTION
**Pull request by lbains**
JIRA: [CPC-496](https://champlainsaintlambert.atlassian.net/browse/CPC-496?atlOrigin=eyJpIjoiMjBkMjA0ZTEyZWE1NDUyMjlhN2Q4MWIxNmZhYzUwN2IiLCJwIjoiaiJ9)
## Context:
In some of our features with UI, we change or edit information about 1 visit only. Instead of reloading information about every visit, It would probably be best if we request the 1 visit instead of all of them.
## Changes
- Made tests for the new code that will be written in the 3 layers.
- Added a new mapping in the Presentation Layer to request 1 visit 
- Added required methods in the business layer

### CODE REVIEW

File: visits-service/src/main/java/com/petclinic/visits/businesslayer/VisitsServiceImpl.java [hunk 1]
```diff
@@ -56,6 +57,17 @@ public Visit addVisit(Visit visit) {
         return visitRepository.findByPetId(petId);
     }
 
+    @Override
+    public Visit getVisitById(int visitId) {
+        if (visitId < 0)
+            throw new InvalidInputException("VisitId can't be negative");
+
+        if (visitId == 33)
+            throw new NotFoundException("Visit with visit id: 33 does not exist.");
```
[the remark this answers is not in the document]
Comment (lbains) [reply visits-service/src/main/java/com/petclinic/visits/businesslayer/VisitsServiceImpl.java]: Done

**Review by bthorng — APPROVED** [review]
LGTM!

File: visits-service/src/test/java/com/petclinic/visits/datalayer/PersistenceTests.java [hunk 2]
```diff
@@ -110,6 +110,18 @@ public void Is_Visit_Empty_Dont_Delete() {
         assertEquals(repo.findByPetId(1).size(), 2);
     }
 
+    @Test
+    public void getVisitByVisitID() {
+        Visit visit = new Visit(5, new Date(), "Description", 5);
+        visit = repo.save(visit);
+
+        Visit foundVisit = repo.findById(visit.getId()).get();
```
Comment (cjayneb) [inline visits-service/src/test/java/com/petclinic/visits/datalayer/PersistenceTests.java:118]: I don't know if the .get() is necessary

**Review by cjayneb — APPROVED** [review]
Looks good for now. I will have to modify it to use VisitDTO with the new public visitId.

**Pull request merged**